hbase-server/src/main/java/org/apache/hadoop/hbase/wal/AbstractFSWALProvider.java

```diff
@@ -500,6 +499,39 @@ public static Path getArchivedLogPath(Path path, Configuration conf) throws IOEx
     }
   }
 
+  /**
+   * Get the archived WAL file path
+   * @param path - active WAL file path
+   * @param conf - configuration
+   * @return archived path if exists, path - otherwise
+   * @throws IOException exception
+   */
+  public static Path getArchivedLog(Path path, Configuration conf) throws IOException {
```

Comment:
Better name it findArchivedLog.
The above getArchivedLogPath is used when we want to move a file into the archived directory, we need to check the configuration to see if we want to put the log into the separate directory or not.
And for this method, it is used when we want to read a wal file, so we need to try all the places to see if we can find it, as we may change the way on how to archive the log on the fly(not sure but in general we want to support onConfigurationChange for more and more configurations).
So name it findArchivedLog will be better, and maybe we could just return null, or return an Optional object when we can not find the wal file.

Reply:
> The above getArchivedLogPath is used when we want to move a file into the archived directory, we need to check the configuration to see if we want to put the log into the separate directory or not.

In production code, these 3 places are where we are calling `getArchivedLogPath` util method.
1. [AbstractFSWALProvider#openReader](https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/wal/AbstractFSWALProvider.java#L524) 
2. [ReplicationSource#getFileSize](https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java#L399)
3. [WALInputFormat.WALRecordReader#nextKeyValue](https://github.com/apache/hbase/blob/master/hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/WALInputFormat.java#L220)
@Apache9  All of the above calls  are trying to find the log in archive path after they couldn't locate the wal in walsDir and they are not used for moving a log file to archive directory. Correct me if I am wrong.

[AbstractFSWALProvider#getWALArchiveDirectoryName](https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/wal/AbstractFSWALProvider.java#L334) is the mehod we use to locate the wal archive directory to move the wal.

Reply:
Good. I think you have found potential bugs in our code base. We should change them all to make use of the new method you introduced in this PR. Suggest opening a new PR to address the incorrect usages of getArchivedLogPath.

Reply:
Cool. Thank you !

Reply:
> Suggest opening a new PR to address the incorrect usages of getArchivedLogPath.

Created https://issues.apache.org/jira/browse/HBASE-26106 for this.